android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootHelper.kt

```diff
@@ -59,8 +59,10 @@ class RNGestureHandlerRootHelper(private val context: ReactContext, wrappedView:
     override fun onHandle(event: MotionEvent, sourceEvent: MotionEvent) {
       val currentState = state
       if (currentState == STATE_UNDETERMINED) {
-        begin()
-        shouldIntercept = false
+        if (!shouldIntercept || orchestrator?.isAnyHandlerActive() != true) {
+          begin()
+          shouldIntercept = false
+        }
```

Comment:
I would probably combine this if with the outer one using `and`, but this also looks ok.